extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeImage.php

```diff
@@ -107,11 +107,13 @@ private function isVideo( $file ) {
 	 * @desc add addtional data required for video media type
 	 * @param array $data
 	 * @param File $file
+	 * @param Title $title
 	 * @return array
 	 */
-	private function videoDataDecorator( $data, $file ) {
-		$data['isVideo'] = true;
-		$data['duration'] = WikiaFileHelper::formatDuration( $file->getMetadataDuration());
+	private function videoDataDecorator( $data, $file, $title ) {
+		$data[ 'isVideo' ] = true;
+		$data[ 'duration' ] = WikiaFileHelper::formatDuration( $file->getMetadataDuration());
+		$data[ 'url' ] = $title->getFullURL();
```

Comment:
You can get title object from $file->getTitle(), however you should check if title is not empty.